# Meson wrapdb's meson.build file doesn't handle installing the library, we patch that here

diff --git a/meson.build b/imgui-1.81-meson.build
index 3343dd4ad..804f84227 100644
--- a/meson.build
+++ b/imgui-1.81-meson.build
@@ -112,8 +112,38 @@
   gnu_symbol_visibility: 'hidden',
   dependencies: dependencies,
   include_directories: include_dirs,
+  version: meson.project_version(),
+  install: true
 )
 
+pkg_mod = import('pkgconfig')
+pkg_mod.generate(imgui,
+  description : 'Dear ImGui: Bloat-free Graphical User interface for C++ with minimal dependencies'
+)
+
+install_headers(
+    'imconfig.h',
+    'imgui.h',
+    'imgui_internal.h',
+    'imstb_rectpack.h',
+    'imstb_textedit.h',
+    'imstb_truetype.h',
+    'backends/imgui_impl_dx9.h',
+    'backends/imgui_impl_osx.h',
+    'backends/imgui_impl_dx10.h',
+    'backends/imgui_impl_dx11.h',
+    'backends/imgui_impl_dx12.h',
+    'backends/imgui_impl_glfw.h',
+    'backends/imgui_impl_glut.h',
+    'backends/imgui_impl_wgpu.h',
+    'backends/imgui_impl_metal.h',
+    'backends/imgui_impl_win32.h',
+    'backends/imgui_impl_vulkan.h',
+    'backends/imgui_impl_opengl2.h',
+    'backends/imgui_impl_opengl3.h',
+    'backends/imgui_impl_allegro5.h',
+    subdir: 'imgui')
+
 if host_machine.system() == 'windows'
   api = '-DIMGUI_API=@0@'.format(get_option('default_library') != 'static' ? '__declspec(dllimport)' : '')
 endif
